### PR TITLE
Return validation errors from api

### PR DESCRIPTION
**Pull request by rkondratowicz**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/model/errors/FieldError.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.reform.jobscheduler.model.errors;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class FieldError {
+
+    @JsonProperty("field_name")
+    public final String fieldName;
+
+    public final String value;
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/model/errors/FieldError.java]: Should this field be named `errorMessage`?

Needs `@JsonProperty` annotation.

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/model/errors/ModelValidationError.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.jobscheduler.model.errors;
+
+import java.util.List;
+
+public class ModelValidationError {
+
+    public final List<FieldError> errors;
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/model/errors/ModelValidationError.java]: Doesn't this field need `@JsonProperty` annotation?

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/ResponseExceptionHandler.java [hunk 3]
```diff
@@ -23,6 +30,24 @@ protected void activateDirectFieldAccess(DataBinder dataBinder) {
         dataBinder.initDirectFieldAccess();
     }
 
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/ResponseExceptionHandler.java:41]: We need to add tests verifying the response. This also gives us a good opportunity to cover all validated fields and check if the response mentions each of the invalid ones.

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/model/errors/FieldError.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/jobscheduler/model/errors/FieldError.java]: renamed to `message`

**Review by lgonczar — APPROVED** [review]

**Pull request merged**